src/WebJobs.Extensions.DurableTask/LifeCycleNotificationHelper.cs

```diff
@@ -0,0 +1,243 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    internal class LifeCycleNotificationHelper
+    {
+        private readonly DurableTaskExtension config;
+        private readonly ExtensionConfigContext extensionConfigContext;
+        private readonly string eventGridKeyValue;
+        private static HttpClient httpClient = null;
+
+        public LifeCycleNotificationHelper(DurableTaskExtension config, ExtensionConfigContext extensionConfigContext)
+        {
+            this.config = config ?? throw new ArgumentNullException(nameof(config));
+            this.extensionConfigContext = extensionConfigContext ?? throw new ArgumentNullException(nameof(extensionConfigContext));
+
+            if (!string.IsNullOrEmpty(config.EventGridTopicEndpoint) && !string.IsNullOrEmpty(config.EventGridKeySettingName))
+            {
+                this.UseTrace = true;
+
+                // Currently, we support Event Grid Custom Topic for notify the lifecycle event of an orchestrator.
+                // For more detail about the Event Grid, please refer this document.
+                // Post to custom topic for Azure Event Grid
+                // https://docs.microsoft.com/en-us/azure/event-grid/post-to-custom-topic
+                httpClient = new HttpClient();
+                INameResolver nameResolver = extensionConfigContext.Config.GetService<INameResolver>();
+                this.eventGridKeyValue = nameResolver.Resolve(config.EventGridKeySettingName);
+                httpClient.DefaultRequestHeaders.Add("aeg-sas-key", this.eventGridKeyValue);
+            }
+        }
+
+        private bool UseTrace { get; }
```

Comment:
Let's make this a `private readonly bool useTrace` instance variable instead of a property.

Reply:
Thank you! We will fix it.